napari_plugin_manager/qt_plugin_dialog.py

```diff
@@ -133,23 +136,29 @@ def __init__(
             self.action_button.setObjectName("install_button")
             self.info_widget.hide()
             self.install_info_button.addWidget(self.info_choice_wdg)
-            self.install_info_button.setFixedWidth(170)
             self.info_choice_wdg.show()
 
-        self._populate_version_dropdown(self.get_installer_source())
-
     def _handle_npe2_plugin(self, npe_version):
         if npe_version in (None, 1):
             return
+
         opacity = 0.4 if npe_version == 'shim' else 1
-        lbl = trans._('npe1 (adapted)') if npe_version == 'shim' else 'npe2'
-        npe2_icon = QLabel(self)
+        text = trans._('npe1 (adapted)') if npe_version == 'shim' else 'npe2'
         icon = QColoredSVGIcon.from_resources('logo_silhouette')
-        npe2_icon.setPixmap(
-            icon.colored(color='#33F0FF', opacity=opacity).pixmap(20, 20)
+        self.set_status(
+            icon.colored(color='#33F0FF', opacity=opacity).pixmap(20, 20), text
         )
-        self.row1.insertWidget(2, QLabel(lbl))
-        self.row1.insertWidget(2, npe2_icon)
+
+    def set_status(self, icon=None, text=''):
```

Comment:
Fixes 

6. refactor code, one example is cleaning up the code that toggles which widgets are visible or not:
https://github.com/napari/napari/pull/5198#discussion_r1028817033